### PR TITLE
Unify telemetry driver for horus and nv14 targets for v2.8 branch, fix 2861

### LINKED ISSUE #2861: EL18 cannot achieve 5.25M BAUD for CRSF
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

Can only achieve 1.87M BAUD, error occurs when trying to use 3.75M BAUD or 5.25M BAUD.

### Expected Behavior

Hardware is similar to TX16S, both uses USART2, it should works the same way as TX16S

### Steps To Reproduce

1. External Module
2. Select CRSF
3. Select BAUD to 3.75M BAUD or 5.35M BAUD
4. ELRS will report connection problem.

### Version

2.8.0

### Transmitter

Flysky EL18

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by richardclli**
Fixes #2861

Summary of changes:
This PR is for 2.8 branch only, there is https://github.com/EdgeTX/edgetx/pull/2862 is for main.

### CODE REVIEW

**Comment by pfeerick** [comment]
This varies somewhat from #2862 for main... is this due to post 2.8 changes, as well as skipping changes to Taranis?

**Comment by raphaelcoeffic** [comment]
I'd say we can keep the changes for 2.8 if it is fixing issues. However, for `main`/`2.9`, we should have the re-written module API which also gets rid of all target specific telemetry drivers, so that #2862 can probably be closed in favour of #3055.

**Comment by richardclli** [comment]
@pfeerick So please merge this one as well.  Why I leave out Taranis is just for simplicity, anyway this need to be reworked in main branch.

**Comment by pfeerick** [comment]
Agreed. Will test with a mind to merge within next 48 hours

File: radio/src/targets/common/arm/stm32/telemetry_driver.cpp [hunk 1]
```diff
@@ -45,17 +45,19 @@ static void telemetryInitDirPin()
   GPIO_InitStructure.GPIO_OType = GPIO_OType_PP;
   GPIO_InitStructure.GPIO_Pin   = TELEMETRY_DIR_GPIO_PIN;
   GPIO_Init(TELEMETRY_DIR_GPIO, &GPIO_InitStructure);
-  GPIO_ResetBits(TELEMETRY_DIR_GPIO, TELEMETRY_DIR_GPIO_PIN);
+  TELEMETRY_DIR_INPUT();
 }
 
-void telemetryPortInit(uint32_t baudrate, uint8_t mode)
+void telemetryPortInitCommon(uint32_t baudrate, uint8_t mode, uint8_t noinv = 0)
```
Comment (pfeerick) [inline radio/src/targets/common/arm/stm32/telemetry_driver.cpp:51]: In future, I would suggest camelCase for the variable names like noInv, and using a define to make the 0/1 parameter have meaning, as well as then being what is compared with in later tests... i.e., at line 90.

**Comment by richardclli** [comment]
Fixed the TX16S problem, will test out in the radio when I get back home.  I think I messed up the definitions of telemetry pin direction, horus and nv14 are different.

File: radio/src/targets/common/arm/stm32/telemetry_driver.cpp [hunk 2]
```diff
@@ -179,6 +205,7 @@ void telemetryPortInvertedInit(uint32_t baudrate)
     case 115200:
       bitLength = 17;
       probeTimeFromStartBit = 23; //because pin is not probed immediately
+//      probeTimeFromStartBit = 25; // TODO: Tarnis used 25, how to merge?
```
Comment (raphaelcoeffic) [inline radio/src/targets/common/arm/stm32/telemetry_driver.cpp:208]: This timing basically depends on interrupt latency until this code is reached. I believe this has been tuned by hand by either @MikeBland or @bsongis. If I have to pick, I'd leave 23, as the latency went up, not down, in general.

**Comment by richardclli** [comment]
I have test the change with my TX16S, looks fine now.

**Comment by pfeerick** [comment]
Yup, same here. Thanks for the confirmation :)

**Pull request merged**